src/lib/Search/PagerContentToDataMapperAbstract.php

```diff
@@ -0,0 +1,136 @@
+    private $userLanguagePreferenceProvider;
+
+    /** @var \eZ\Publish\API\Repository\LanguageService */
+    private $languageService;
+
+    /**
+     * @param \eZ\Publish\API\Repository\ContentTypeService $contentTypeService
+     * @param \eZ\Publish\API\Repository\UserService $userService
+     * @param \eZ\Publish\Core\MVC\Symfony\Locale\UserLanguagePreferenceProviderInterface $userLanguagePreferenceProvider
+     * @param \eZ\Publish\Core\Helper\TranslationHelper $translationHelper
+     * @param \eZ\Publish\API\Repository\LanguageService $languageService
+     */
+    public function __construct(
+        TranslationHelper $translationHelper,
+        ContentTypeService $contentTypeService,
+        UserService $userService,
+        UserLanguagePreferenceProviderInterface $userLanguagePreferenceProvider,
+        LanguageService $languageService
+    ) {
+        $this->contentTypeService = $contentTypeService;
+        $this->userService = $userService;
+        $this->userLanguagePreferenceProvider = $userLanguagePreferenceProvider;
+        $this->translationHelper = $translationHelper;
+        $this->languageService = $languageService;
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     * @param bool $filterDisabled
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\Language[]
+     */
+    protected function getAvailableTranslations(
+        Content $content,
+        bool $filterDisabled = false
+    ): iterable {
+        $availableTranslationsLanguages = $this->languageService->loadLanguageListByCode(
+            $content->versionInfo->languageCodes
+        );
+
+        if (false === $filterDisabled) {
+            return $availableTranslationsLanguages;
+        }
+
+        return array_filter(
+            $availableTranslationsLanguages,
+            (static function (Language $language): bool {
+                return $language->enabled;
+            })
+        );
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Content $content
+     *
+     * @return bool
+     */
+    protected function isContentIsUser(Content $content): bool
+    {
+        return (new ContentIsUser($this->userService))->isSatisfiedBy($content);
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     *
+     * @return \eZ\Publish\API\Repository\Values\User\User|null
+     */
+    protected function getContributor(ContentInfo $contentInfo): ?User
+    {
+        $userExists = (new UserExists($this->userService))->isSatisfiedBy($contentInfo->ownerId);
+
+        if (false === $userExists) {
+            return null;
+        }
+
+        return $this->userService->loadUser($contentInfo->ownerId);
+    }
+
+    /**
+     * @param array $data
+     * @param array $contentTypeIds
+     */
+    protected function setTranslatedContentTypesNames(array &$data, array $contentTypeIds): void
```

Comment:
not rly fan of passing array by reference, but if no one protests ;)

Reply:
I understand :) also not a rly fan of this method, but it's only moved from another place with little refactor (not more time in this task)